src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -603,6 +610,26 @@ impl AccessChecker for PrivilegeAccess {
                     .await?;
             }
             Plan::CopyIntoTable(plan) => {
+                match plan.stage_table_info.stage_info.stage_type {
+                    StageType::External => {
+                        session
+                            .validate_privilege(
+                                &GrantObject::Global,
```

Comment:
IMO the privileges around reading/writing a stage is expected to be bound with the target Stage instead of a Global privilege ?

Reply:
Ok, looks like we need to 

1. add new GrantObject stage and udf. 
2. grant need to check the grant object.

So let use set this pr to draft.